generator/wrapper_gen.py

```diff
@@ -175,9 +175,13 @@ def my_procid():
 
 def _get_data(x):
     if isinstance(x, pdDataFrame):
-        x = [x.loc[:,i].values for i in x]
+        x_dtypes = x.dtypes.values
+        if np.all(x_dtypes == x_dtypes[0]):
+            x = x.to_numpy()
+        else:
+            x = [xi.to_numpy() for _, xi in x.items()]
     elif isinstance(x, pdSeries):
-        x = [x.values]
+        x = [x.to_numpy()]
```

Comment:
I do not see how this solves the problem. 

```
X = np.random.randn(10,3)
ps = pd.Series(X[:, 0])
```

and now

```
In [57]: ps.to_numpy().flags
Out[57]:
  C_CONTIGUOUS : False
  F_CONTIGUOUS : False
  OWNDATA : False
  WRITEABLE : True
  ALIGNED : True
  WRITEBACKIFCOPY : False
  UPDATEIFCOPY : False

```

and we hit the trouble-some branch, which expects C-contiguous array, but gets a non-contiguous one.

The fix must implement handling of non-contiguous arrays in `make_nt`.

Reply:
This was exactly my suspicion. 
A short-term fix would make a copy. A proper solution would be able to handle SOA of non-contiguous arrays. We currently only support homogen non-contiguous data.

Reply:
And yes, this will most likely degrade performance over just making a copy. See above argument on pros and cons of making copies vs operating in-place.

Reply:
@fschlimb @oleksandr-pavlyk  Thank you for your comment! Fixed, the current behavior will go to the `make_nt` in the case non-contiguous arrays.